frontend/src/metabase/admin/settings/components/CloudPanel/CloudPanel.tsx

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useState } from "react";
+import { t } from "ttag";
+
+import {
+  useGetCloudMigrationQuery,
+  useCreateCloudMigrationMutation,
+} from "metabase/api";
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+import { useDispatch } from "metabase/lib/redux";
+import { refreshSiteSettings } from "metabase/redux/settings";
+import { Box, Text } from "metabase/ui";
+
+import { MigrationError } from "./MigrationError";
+import { MigrationInProgress } from "./MigrationInProgress";
+import { MigrationStart } from "./MigrationStart";
+import { MigrationSuccess } from "./MigrationSuccess";
+import {
+  type InternalCloudMigrationState,
+  getCheckoutUrl,
+  isInProgressMigration,
+} from "./utils";
+import { getStartedVisibleStates, pollingIntervalsByState } from "./utils";
+
+export const CloudPanel = () => {
+  const dispatch = useDispatch();
+  const [pollingInterval, setPollingInterval] = useState<number | undefined>(
+    undefined,
+  );
+
+  const {
+    data: migration,
+    isLoading,
+    error,
+  } = useGetCloudMigrationQuery(undefined, {
+    refetchOnMountOrArgChange: true,
+    pollingInterval,
+  });
+
+  const migrationState: InternalCloudMigrationState =
+    migration?.state ?? "uninitialized";
+
+  useEffect(
+    function syncPollingInterval() {
+      setPollingInterval(pollingIntervalsByState[migrationState]);
+    },
+    [migrationState],
+  );
+
+  useEffect(
+    function syncSiteSettings() {
+      if (migrationState) {
+        dispatch(refreshSiteSettings({}));
+      }
+    },
+    [dispatch, migrationState],
+  );
+
+  const [createCloudMigration] = useCreateCloudMigrationMutation();
+
+  const handleCreateMigration = async () => {
+    const migration = await createCloudMigration().unwrap();
```

Comment:
IMHO, to avoid creating multiple migrations, we should "block" the UI (read only) straight away after clicking the start migration button, before any requests happen.

Reply:
We can probably put something together for that, but in reality if we're truly concerned about creating multiple migrations, the BE will also need to ensure that it's not possible to create multiple migrations